libs/wingsdk/src/std/map.ts

```diff
@@ -0,0 +1,70 @@
+// These classes are used by Wing to provide JSII subsets of the JS Array class.
+// They should not be consumed directly by users.
+// TODO: These should be interfaces, currently Wing does not support interface JSII imports
+
+/**
+ * Immutable Map
+ */
+export class ImmutableMap {
+  /**
+   * Returns the number of elements in the map.
+   */
+  public get size(): number {
+    throw new Error("Abstract");
+  }
+
+  /**
+   * Returns a specified element from the map. If the value that is associated to the provided key
+   * is an object, then you will get a reference to that object and any change made to that object
+   * will effectively modify it inside the map.
```

Comment:
(thinking about docs) What counts as an "object" in Wing? (we want to make sure we are not leaking javascript impl details to the user)

Reply:
I think this is okay for now...